packages/graph/hash_graph/lib/graph/src/store/postgres/query/statement/select.rs

```diff
@@ -49,3 +49,168 @@ impl Transpile for SelectStatement<'_> {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::borrow::Cow;
+
+    use postgres_types::ToSql;
+    use type_system::DataType;
+
+    use crate::{
+        ontology::DataTypeQueryPath,
+        store::{
+            postgres::query::{test_helper::trim_whitespace, PostgresQueryRecord, SelectCompiler},
+            query::{Filter, FilterExpression, Parameter},
+        },
+    };
+
+    fn test_compilation<'f: 'q, 'q, T: PostgresQueryRecord<'q>>(
+        compiler: &SelectCompiler<'f, 'q, T>,
+        expected_statement: &'static str,
+        expected_parameters: &[&'f dyn ToSql],
+    ) {
+        let (compiled_statement, compiled_parameters) = compiler.compile();
+
+        assert_eq!(
+            trim_whitespace(compiled_statement),
+            trim_whitespace(expected_statement)
+        );
+
+        let compiled_parameters = compiled_parameters
+            .iter()
+            .map(|parameter| format!("{parameter:?}"))
+            .collect::<Vec<_>>();
+        let expected_parameters = expected_parameters
+            .iter()
+            .map(|parameter| format!("{parameter:?}"))
+            .collect::<Vec<_>>();
+
+        assert_eq!(compiled_parameters, expected_parameters);
+    }
+
+    #[test]
+    fn asterix() {
+        test_compilation(
+            &SelectCompiler::<DataType>::with_asterix(),
```

Comment:
Have we renamed `asterix` yet? :rofl:

Reply:
🤦🏻‍♂️ Should be fixed now 31906a75cab97d1ed6335e3211045d100cb87176